src/components/DiffView/index.spec.tsx

```diff
@@ -222,6 +225,196 @@ describe(__filename, () => {
     expect(_document.querySelector).toHaveBeenCalledWith(location.hash);
   });
 
+  it('tries to find the selected line element on update', () => {
+    const _document = {
+      ...document,
+      querySelector: jest.fn(),
+    };
+    const location = createFakeLocation({ hash: '#some-hash' });
+
+    const root = render({ _document });
+    root.setProps({ location });
+
+    expect(_document.querySelector).toHaveBeenCalledWith(location.hash);
+  });
+
+  it('calls history.push to navigate to the first diff when showDiff is "first"', () => {
+    const firstDiffAnchor = 'D5';
+    const _getRelativeDiffAnchor = jest.fn().mockReturnValue(firstDiffAnchor);
+    const location = createFakeLocation({
+      search: queryString.stringify({ showDiff: DiffPosition.first }),
+    });
+    const history = createFakeHistory({ location });
+
+    const queryParams = queryString.parse(location.search);
+    const newParams = { ...queryParams };
+
+    delete newParams.showDiff;
+    const newLocation = {
+      ...location,
+      hash: `#${firstDiffAnchor}`,
+      search: `?${queryString.stringify(newParams)}`,
+    };
+
+    renderWithLinterProvider({ _getRelativeDiffAnchor, history, location });
+
+    expect(history.push).toHaveBeenCalledWith(newLocation);
+  });
+
+  it('calls history.push to navigate to the last diff when showDiff is "last"', () => {
+    const lastDiffAnchor = 'D5';
+    const _getDiffAnchors = jest
+      .fn()
+      .mockReturnValue(['D1', 'D2', lastDiffAnchor]);
+    const location = createFakeLocation({
+      search: queryString.stringify({ showDiff: DiffPosition.last }),
+    });
+    const history = createFakeHistory({ location });
+
+    const queryParams = queryString.parse(location.search);
+    const newParams = { ...queryParams };
+
+    delete newParams.showDiff;
+    const newLocation = {
+      ...location,
+      hash: `#${lastDiffAnchor}`,
+      search: `?${queryString.stringify(newParams)}`,
```

Comment:
I am not sure you need to prepend the stringified query string with a `?`, see: https://github.com/sindresorhus/query-string#usage